### PR TITLE
Added a fix for nil issue

### PR DESCRIPTION
**Pull request by amulyavarote**
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

# Description

Added a nil check to avoid the nil error.

## Issue reference

Please reference the issue this PR will close: #4527 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by artursouza** [comment]
Please, change or create a new PR to merge into release-1.7 and I will cut a new RC for the 1.7.1 patch release.

**Review by artursouza — DISMISSED** [review]

**Comment by yaron2** [comment]
Why is this going into 1.7?

File: pkg/grpc/server.go [hunk 1]
```diff
@@ -314,7 +314,9 @@ func shouldRenewCert(certExpiryDate time.Time, certDuration time.Duration) bool
 
 func (s *server) getGRPCAPILoggingInfo() grpc_go.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc_go.UnaryServerInfo, handler grpc_go.UnaryHandler) (interface{}, error) {
-		s.infoLogger.Info("gRPC API Called: ", *info)
+		if info != nil {
```
Comment (yaron2) [inline pkg/grpc/server.go:317]: Needs explanation on why this is part of the hotfix. I see this was labeled as `P1` which I agree with. We don't include commits normally for `P1` issues in a patch release.

Comment (amulyavarote) [reply pkg/grpc/server.go:317]: According to me, it is breaking the functionality(enable-api-logging) that we added in 1.7. So it needs to go in to 1.7.

Comment (yaron2) [reply pkg/grpc/server.go:317]: Thanks for explaining @amulyavarote, we'd like to have maintainer consensus (2/3 agreeing in this case) before merging. I'll discuss with @artursouza and/or @daixiang0 to determine this.

Comment (amulyavarote) [reply pkg/grpc/server.go:317]: Sure. Thanks @yaron2

Comment (artursouza) [reply pkg/grpc/server.go:317]: As discussed offline, @yaron2 and I agreed on this being merged.

Comment (artursouza) [reply pkg/grpc/server.go:317]: I am creating a new PR containing the doc change.

**Review by artursouza — APPROVED** [review]

**Comment by yaron2** [comment]
Good for 1.7.1.

**Pull request merged**